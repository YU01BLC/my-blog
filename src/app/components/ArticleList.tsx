'use client';

import { useMediaQuery } from '@mui/material';
import Box from '@mui/material/Box';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import Link from 'next/link';

export default function ArticleList() {
  const array = [1, 2, 3];
  const mediaView: boolean = useMediaQuery('(min-width:1024px)');
  return (
    <Box data-testid='article-list'>
      <Link href='/hoge'>
        {array.map((index) => (
          <Card key={index} sx={{ marginBottom: '2em' }}>
            <CardHeader title={`記事タイトル: ${index}`} subheader='2023/12/1' />

            <CardMedia
              component='img'
              height={mediaView ? '150' : '500'}
              image={`https://source.unsplash.com/collection/1346951/1000x500?sig=${index}`}
              alt={`blog image: ${index}`}
            />
            <CardContent>
              <Typography
                variant='body1'
                color='text.secondary'
                sx={{ paddingTop: '1em' }}
                data-testid={`article-list--text${index}`}
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Veritatis quod maiores totam aperiam
                necessitatibus est, exercitationem architecto quibusdam accusamus maxime asperiores eaque adipisci
                itaque nemo, minus odio, alias dolor possimus!
              </Typography>
            </CardContent>
          </Card>
        ))}
      </Link>
    </Box>
  );
}
