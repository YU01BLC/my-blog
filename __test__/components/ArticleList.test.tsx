import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import ArticleList from '../../src/app/components/ArticleList';

describe('ArticleListコンポーネントSnapshot', () => {
  it('snapshot', () => {
    const { asFragment } = render(<ArticleList />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('ArticleListコンポーネント表示確認', () => {
  it('ArticleListコンポーネントが表示されていること', () => {
    render(<ArticleList />);
    const element = screen.getByTestId('article-list');
    expect(element).toBeInTheDocument();
  });
});

describe('Cardコンポーネント表示確認', () => {
  const cardData = [
    { id: 1, title: '記事タイトル: 1', alt: 'blog image: 1', testId: 'article-list--text1' },
    { id: 2, title: '記事タイトル: 2', alt: 'blog image: 2', testId: 'article-list--text2' },
    { id: 3, title: '記事タイトル: 3', alt: 'blog image: 3', testId: 'article-list--text3' },
  ];
  it('CardHeaderコンポーネントが表示されていること', () => {
    render(<ArticleList />);
    cardData.forEach((card) => {
      // getByTitleで取得するとCSS FWの仕様上取得できない(title要素として表示されていない)
      const cardElement = screen.getByText(card.title);
      expect(cardElement).toBeInTheDocument();
    });
  });
  it('CardMediaコンポーネントが表示されていること', () => {
    render(<ArticleList />);
    cardData.forEach((card) => {
      // getByTitleで取得するとCSS FWの仕様上取得できない(title要素として表示されていない)
      const cardElement = screen.getByAltText(card.alt);
      expect(cardElement).toBeInTheDocument();
    });
  });
  it('Typographyコンポーネントが表示されていること', () => {
    render(<ArticleList />);
    cardData.forEach((card) => {
      const cardElement = screen.getByTestId(card.testId);
      expect(cardElement).toBeInTheDocument();
    });
  });
});

describe('Cardコンポーネント動作確認テスト', () => {
  // app routerでのページ遷移に関するテスト方法が現状不明。一旦スキップで対応する。
  xit('Cardコンポーネント内の画像箇所押下で「hoge」ページに遷移すること', async () => {
    render(<ArticleList />);
    userEvent.click(screen.getByAltText(/Paella dish/i));
    expect(await screen.getByText('')).toBeInTheDocument();
  });
});
