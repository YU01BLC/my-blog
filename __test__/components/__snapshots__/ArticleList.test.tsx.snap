// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListコンポーネントSnapshot snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
    data-testid="article-list"
  >
    <a
      href="/hoge"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5y2sye-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              記事タイトル: 1
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
            >
              2023/12/1
            </span>
          </div>
        </div>
        <img
          alt="blog image: 1"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="500"
          src="https://source.unsplash.com/collection/1346951/1000x500?sig=1"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
            data-testid="article-list--text1"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Veritatis quod maiores totam aperiam necessitatibus est, exercitationem architecto quibusdam accusamus maxime asperiores eaque adipisci itaque nemo, minus odio, alias dolor possimus!
          </p>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5y2sye-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              記事タイトル: 2
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
            >
              2023/12/1
            </span>
          </div>
        </div>
        <img
          alt="blog image: 2"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="500"
          src="https://source.unsplash.com/collection/1346951/1000x500?sig=2"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
            data-testid="article-list--text2"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Veritatis quod maiores totam aperiam necessitatibus est, exercitationem architecto quibusdam accusamus maxime asperiores eaque adipisci itaque nemo, minus odio, alias dolor possimus!
          </p>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5y2sye-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              記事タイトル: 3
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
            >
              2023/12/1
            </span>
          </div>
        </div>
        <img
          alt="blog image: 3"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="500"
          src="https://source.unsplash.com/collection/1346951/1000x500?sig=3"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
            data-testid="article-list--text3"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Veritatis quod maiores totam aperiam necessitatibus est, exercitationem architecto quibusdam accusamus maxime asperiores eaque adipisci itaque nemo, minus odio, alias dolor possimus!
          </p>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
